.github/workflows/certora_recovery.yml

```diff
@@ -8,8 +8,30 @@ on:
     branches:
       - main
 
+env:
+  CERTORAKEY: ${{ secrets.CERTORA_KEY }}
+
 jobs:
+  check-secret:
+    runs-on: ubuntu-latest
+    outputs:
+      certora-key-exists: ${{ steps.certora-key-check.outputs.defined }}
+    steps:
+      - name: Check for Certora Secret availability
+        id: certora-key-check
+        # perform secret check & put boolean result as an output
+        run: |
+          if [[ -n "${CERTORAKEY}" ]]; then
+            echo "defined=true" >> $GITHUB_OUTPUT;
+            echo "CERTORA_KEY exists"
+          else
+            echo "defined=false" >> $GITHUB_OUTPUT;
+            echo "CERTORA_KEY does not exist"
+          fi
+
   verify:
+    needs: [check-secret]
+    if: needs.check-secret.outputs.certora-key-exists == 'true'
```

Comment:
Shouldn't this be:

```suggestion
    if: needs.check-secret.outputs.defined == 'true'
```

At least that is my understanding of how `GITHUB_OUTPUT` works.

Reply:
I believe the current code is correct based on the suggestion

<img width="697" alt="Screenshot 2024-07-01 at 10 51 11 AM" src="https://github.com/5afe/CandideWalletContracts/assets/30735581/12902f9f-d38e-4278-80fa-6066ed2ec16c">

And based on the example mentioned in [here](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-setting-an-output-parameter).

Reply:
Oops, this is what I meant:

https://github.com/5afe/CandideWalletContracts/blob/be2952bee156b5da0c4a7bea71d4554066355841/.github/workflows/certora_recovery.yml#L18

I was just looking at the wrong job :see_no_evil:.

LGTM. Thanks for the clarification!